### PR TITLE
wc: mb: Add source identify function and project command

### PR DESCRIPTION
**Pull request by MouchenHung-QUANTA**
Summary:
- Add source identification between main source and second source.
- Add project command.

Test Plan:
- Build Code: PASS
- Check project info command in BIC console:

Log:
- BIC console:
  ```
  uart:~$ wcmb info
  ========================{PLATFORM INFO}========================
  * NAME:          Waimea Canyon platform command
  * DESCRIPTION:   Commands that could be used to debug or validate.
  * Note:          none
  ------------------------------------------------------------------
  * SYSTEM CLASS:  Dual
  * SYSTEM SOURCE: Sencond
  ========================{PLATFORM INFO}========================
  ```

### CODE REVIEW

File: common/shell/commands/info_shell.c [hunk 1]
```diff
@@ -18,9 +18,9 @@
 #include "plat_version.h"
 
 #ifdef CONFIG_BOARD_AST1030_EVB
-#define BIC_BOARD_NAME "AST1030"
-#elif
-#define BIC_BOARD_NAME "Unknown"
+#define BIC_CHIP_NAME "AST1030"
+#else
+#define BIC_CHIP_NAME "Unknown"
 #endif
```
Comment (GoldenBug) [inline common/shell/commands/info_shell.c:24]: We can use `CONFIG_BOARD` which is defined by Zephyr to get the board we're building for.
This means we don't have to define extra fields.

```
#ifndef CONFIG_BOARD
#define CONFIG_BOARD "unknown"
#endif
.
.
.
shell_print(shell, "* CHIP/OS:       %s - %s", CONFIG_BOARD, RTOS_TYPE);
```

File: meta-facebook/wc-mb/src/shell/commands/plat_info_shell.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "plat_info_shell.h"
+#include "plat_class.h"
+
+int cmd_wcmb_info_print(const struct shell *shell, size_t argc, char **argv)
```
Comment (GoldenBug) [inline meta-facebook/wc-mb/src/shell/commands/plat_info_shell.c:20]: I think we should make the `cmd_info_print (...)` __weak and just override it to include extra information.
I don't think it's necessary to make a different class of commands just for wc.

Instead we can add them to platform commands if we need to add new commands in the future.

File: meta-facebook/wc-mb/src/shell/commands/plat_info_shell.c [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	;
+	shell_print(shell, "========================{PLATFORM INFO}========================");
+	shell_print(shell, "* NAME:          Waimea Canyon platform command");
+	shell_print(shell, "* DESCRIPTION:   Commands that could be used to debug or validate.");
+	shell_print(shell, "* Note:          none");
+	shell_print(shell, "------------------------------------------------------------------");
+	shell_print(shell, "* SYSTEM CLASS:  %s",
+		    get_system_class() == SYS_DUAL ? "Dual" : "Single");
+	shell_print(shell, "* SYSTEM SOURCE: %s",
+		    get_source_class() == SRC_MAIN ? "Main" : "Sencond");
```
Comment (GoldenBug) [inline meta-facebook/wc-mb/src/shell/commands/plat_info_shell.c:31]: Refactor the cmd_info_print function to something like this

```
void print_main_info(const struct shell *shell)
{
        shell_print(
                shell,
                "========================{SHELL COMMAND INFO}========================================");
        shell_print(shell, "* NAME:          Platform command");
        shell_print(shell, "* DESCRIPTION:   Commands that could be used to debug or validate.");
        shell_print(shell, "* DATE/VERSION:  none");
        shell_print(shell, "* CHIP/OS:       %s - %s", CONFIG_BOARD, RTOS_TYPE);
        shell_print(shell, "* Note:          none");
        shell_print(shell, "------------------------------------------------------------------");
        shell_print(shell, "* PLATFORM:      %s-%s", PLATFORM_NAME, PROJECT_NAME);
        shell_print(shell, "* BOARD ID:      %d", BOARD_ID);
        shell_print(shell, "* STAGE:         %d", PROJECT_STAGE);
        shell_print(shell, "* FW VERSION:    %d.%d", FIRMWARE_REVISION_1, FIRMWARE_REVISION_2);
        shell_print(shell, "* FW DATE:       %x%x.%x.%x", BIC_FW_YEAR_MSB, BIC_FW_YEAR_LSB,
                ⦙   BIC_FW_WEEK, BIC_FW_VER);
        shell_print(shell, "* FW IMAGE:      %s.bin", CONFIG_KERNEL_BIN_NAME);
        shell_print(
                shell,
                "========================{SHELL COMMAND INFO}========================================");
}

__weak int cmd_info_print(const struct shell *shell, size_t argc, char **argv)
{
        print_main_info(shell);
        return 0;
}
```

Then we can override this function and call the same print_main_info and then print the additional system_class info.

File: meta-facebook/wc-mb/src/shell/plat_shell.c [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/*
```
Comment (GoldenBug) [inline meta-facebook/wc-mb/src/shell/plat_shell.c:1]: no need to add a different type of command just for this system

**Review by GoldenBug — CHANGES_REQUESTED** [review]

**Comment by MouchenHung-QUANTA** [comment]
Hi @GoldenBug ,

New commit has been uploaded, please have a look, thanks.

**Pull request closed without merging**